src/sys/windows/selector.rs

```diff
@@ -194,9 +257,15 @@ impl Registration {
         if self.selector.is_some() {
             return Err(other("socket already registered"))
         }
+
         try!(Registration::validate_opts(opts));
         try!(selector.inner.port.add_socket(self.token.as_usize(), socket));
         self.associate(selector, token);
+
+        if opts.is_level() {
+            self.key = Some(socket.as_raw_socket() as usize);
```

Comment:
Perhaps this could happen unconditionally? Just helps follow the code at least (either way shouldn't matter)